### PR TITLE
Spherical coordinates

### PR DESCRIPTION
**Pull request by eggrobin**
Support for spherical coordinates. The various implementations of right ascension and declination should be replaced by that in a future CL.

### CODE REVIEW

File: geometry/r3_element.hpp [hunk 1]
```diff
@@ -47,6 +55,26 @@ struct R3Element {
...
+  // obviate confusion over the order of |latitude| and |longitude|.
+  SphericalCoordinates();
+
+  // Uses the x-y plane as the equator, the x-axis as the reference direction on
+  // the equator, and the z-axis as the north pole.
+  R3Element<Scalar> ToCartesian();
+
+  Angle longitude;
+  Angle latitude;
+  Scalar radius;
```
Comment (pleroy) [inline geometry/r3_element.hpp]: Why not in the same order as the parameters of the factory?

File: geometry/r3_element_body.hpp [hunk 2]
```diff
@@ -7,9 +7,14 @@
 #include "quantities/elementary_functions.hpp"
 #include "quantities/quantities.hpp"
 #include "quantities/serialization.hpp"
+#include "r3_element.hpp"
```
Comment (pleroy) [inline geometry/r3_element_body.hpp]: geometry/r3_element.hpp

File: geometry/r3_element_body.hpp [hunk 3]
```diff
@@ -7,9 +7,14 @@
...
 
 namespace principia {
 
+using quantities::ArcSin;
+using quantities::ArcTan;
+using quantities::Cos;
+using quantities::Sin;
 using quantities::DoubleOrQuantitySerializer;
```
Comment (pleroy) [inline geometry/r3_element_body.hpp]: Alphabetize.

File: geometry/r3_element_body.hpp [hunk 4]
```diff
@@ -125,6 +139,28 @@ R3Element<Scalar> R3Element<Scalar>::ReadFromMessage(
 }
 
 template<typename Scalar>
+SphericalCoordinates<Scalar>::SphericalCoordinates() {};
+
+template<typename Scalar>
+R3Element<Scalar> SphericalCoordinates<Scalar>::ToCartesian() {
+  double cos_latitude = Cos(latitude);
```
Comment (pleroy) [inline geometry/r3_element_body.hpp]: const

File: geometry/r3_element_test.cpp [hunk 5]
```diff
@@ -20,17 +22,22 @@ using bipm::Knot;
...
 using quantities::Time;
 using si::Day;
+using si::Degree;
 using si::Hour;
 using si::Kilo;
 using si::Metre;
 using si::Minute;
 using si::Second;
+using testing_utilities::Componentwise;
 using testing_utilities::AlmostEquals;
```
Comment (pleroy) [inline geometry/r3_element_test.cpp]: Alphabetize.

File: geometry/r3_element.hpp (hunk 1 above)
Comment (eggrobin) [inline geometry/r3_element.hpp]: done.

File: geometry/r3_element_body.hpp (hunk 2 above)
Comment (eggrobin) [inline geometry/r3_element_body.hpp]: doen.

File: geometry/r3_element_body.hpp (hunk 3 above)
Comment (eggrobin) [inline geometry/r3_element_body.hpp]: edno

File: geometry/r3_element_body.hpp (hunk 4 above)
Comment (eggrobin) [inline geometry/r3_element_body.hpp]: done.

File: geometry/r3_element_test.cpp (hunk 5 above)
Comment (eggrobin) [inline geometry/r3_element_test.cpp]: Done.

**Pull request merged**